examples/src/main/scala/org/apache/spark/examples/streaming/MQTTWordCount.scala

```diff
@@ -42,25 +40,30 @@ object MQTTPublisher {
     StreamingExamples.setStreamingLogLevels()
 
     val Seq(brokerUrl, topic) = args.toSeq
+    
+    var client: MqttClient = null
 
     try {
-      var peristance:MqttClientPersistence =new MqttDefaultFilePersistence("/tmp")
-      client = new MqttClient(brokerUrl, MqttClient.generateClientId(), peristance)
-    } catch {
-      case e: MqttException => println("Exception Caught: " + e)
-    }
+      val persistence = new MemoryPersistence()
+      client = new MqttClient(brokerUrl, MqttClient.generateClientId(), persistence)
 
-    client.connect()
+      client.connect()
 
-    val msgtopic: MqttTopic = client.getTopic(topic)
-    val msg: String = "hello mqtt demo for spark streaming"
+      val msgtopic  = client.getTopic(topic)
+      val msgContent = "hello mqtt demo for spark streaming"
+      val message = new MqttMessage(msgContent.getBytes("utf-8"))
 
-    while (true) {
-      val message: MqttMessage = new MqttMessage(String.valueOf(msg).getBytes("utf-8"))
-      msgtopic.publish(message)
-      println("Published data. topic: " + msgtopic.getName() + " Message: " + message)
+      while (true) {
+        Thread.sleep(100)
+        msgtopic.publish(message)
+        println("Published data. topic: %s; Message: %s".format(msgtopic.getName(), message))
+      }
+      
+    } catch {
+      case e: MqttException => println("Exception Caught: " + e)
```

Comment:
I don't see the point in swallowing this exception. Why not let it percolate, it will anyway print the stacktrace and stop the process?